packages/openapi-ts/src/plugins/@hey-api/client-fetch/bundle/client.ts

```diff
@@ -106,23 +106,45 @@ export const createClient = (config: Config = {}): Client => {
     };
 
     if (response.ok) {
+      const parseAs =
+        (opts.parseAs === 'auto'
+          ? getParseAs(response.headers.get('Content-Type'))
+          : opts.parseAs) ?? 'json';
+
       if (
         response.status === 204 ||
         response.headers.get('Content-Length') === '0'
       ) {
+        let emptyData: any;
```

Comment:
Did you consider calling `data = await response[parseAs]();` here as well? Would it produce a different result?